friends/src/App.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import { Route } from 'react-router-dom';
+
+import './App.css';
+import FriendsList from './Components/FriendsList';
+import NavBar from './Components/NavBar/NavBar';
+import FriendForm from './Components/FriendForm/FriendForm'
+
+class App extends Component {
+  constructor(){
+    super();
+    this.state= {
+      friends:[]
+    };
+  }
+
+  componentDidMount(){
+   this.getFriendsData()
+  }
+
+ getFriendsData  = ()=>{
+  axios.get('http://localhost:5000/friends')
+  .then((result)=>{
+    // console.log(result);
+    this.setState({friends: result.data})
+  })
+ }
+
+ submitNewFriend = friend =>{
+  axios.post('http://localhost:5000/friends',friend)
+  .then(response=>{
+    console.log(response)
+    this.setState({friends: response.data})
+  })
+  .catch(error=>console.log(error))
+}
+
+ editFriend = (friend,id) =>{
+  axios.put(`http://localhost:5000/friends/${id}`,friend)
+  .then(response=>{
+    //I don't know why but even though state is updated i wont get the new list until I refresh.
+    this.setState({friends: response.data})
```

Comment:
This is happening because you are setting the child's initial state using props, instead of just using the object passed down from props. Because of this, it doesn't think that anything needs to change, as the constructor was already called. The problem is in your Friend.js component. I hope that helps!